Ironmon-Tracker.lua

```diff
@@ -418,6 +420,29 @@ function Main.GenerateNextRom()
 	}
 end
 
+function Main.SaveCurrentRom(filename)
+	if Main.FileExists(filename) then
+		Main.CopyFile(filename, string.format("prev-seed %s", filename))
+		Main.CopyFile(string.format("%s.log", filename), string.format("prev-seed %s.log", filename))
+	end
+end
+
+function Main.CopyFile(filename, newfilename)
+	local original = io.open(filename, "rb")
+	local copy = io.open(newfilename, "wb")
+
+	while true do
+		local originalBlock = original:read(2^13)
+		if not originalBlock then 
+		  break
+		end
+		copy:write(originalBlock)
+	end
```

Comment:
Small request here for readability, but can you move the condition for the loop to the loops definition itself. For example:

> local nextBlock = original:read(2^13)
> while nextBlock ~= nil do
> copy:write(originalBlock)
> nextBlock = original:read(2^13)
> end

I believe this has the same functionality, just a bit cleaner. Worth double-checking it still works.

Reply:
[f38fe80](https://github.com/besteon/Ironmon-Tracker/pull/283/commits/f38fe801fbb91bea7c63fb475d5c29c4dca80a35)

Reply:
also to add to this, may want to perhaps add safety nil checks on `original` and `copy` after their `io.open` calls to confirm they're non-nil before doing anything with them / attempting to close the files

Reply:
I don't see the reason to check for nil when opening (in theory there's no harm in copying to/from empty files) but I added logic to make it safe to copy (And append to)